src/neo-vm/Types/Struct.cs

```diff
@@ -44,6 +47,9 @@ public Struct Clone()
                 Struct b = queue.Dequeue();
                 foreach (StackItem item in b)
                 {
+                    if (count == 0) throw new ArgumentException();
+                    count--;
```

Comment:
It can be an array or map.

Reply:
We can subtract the `SubItemsCount`, but it could have another array inside

Reply:
You've probably seen nspcc-dev/neo-go#2054, I think it's enough to prevent complexity explosion without changing VM behavior (it's possible to return more than `MaxStackSize` items from `Clone` this way, but not a lot more and it'll be checked when item is to be pushed onto the stack).